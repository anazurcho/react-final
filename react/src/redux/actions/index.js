const userAuthenticated = ( email, token ) => {
    // localStorage.setItem( "token:", token );
    // console.log(token)
    return {
        type: 'USER_AUTHENTICATED',
        payload: {
            email: email,
            token: token
        }
    };
}
const userLoggedOut = () => {
    return {
        type: 'USER_LOGGED_OUT',
        token:false
    };
}
const ChangeTheme = ( theme ) => {
    return {
        type: 'CHANGE_THEME',
        payload: theme
    };
}

export {
    userAuthenticated,
    userLoggedOut,
    ChangeTheme,
    USER_AUTHENTICATED,
    CHANGE_THEME,
    USER_LOGGED_OUT

};
const USER_AUTHENTICATED = "USER_AUTHENTICATED";
const CHANGE_THEME = "CHANGE_THEME";
const USER_LOGGED_OUT = "USER_LOGGED_OUT";