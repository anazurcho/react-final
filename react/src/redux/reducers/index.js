const initialState = {
    isLoggedIn: false,
    token: "",
    email:"",
    theme:"darkly"
}

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case 'USER_AUTHENTICATED':
            localStorage.setItem( "token:", action.payload.token );
            return {
                isLoggedIn: true,
                email: action.payload.email,
                token: action.payload.token,
            };
        case 'CHANGE_THEME':
            return{
                ...state,
                theme: action.payload
            };
        case 'USER_LOGGED_OUT':
            localStorage.setItem("token:", false);
            return {
                isLoggedIn: false,
                token: "",
                email:""
            };
        default:
            return state;
    }
}

export default reducer;
