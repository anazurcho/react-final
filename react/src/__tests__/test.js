import React from "react";
import reducer from "../redux/reducers"
import {USER_AUTHENTICATED,USER_LOGGED_OUT, CHANGE_THEME, ChangeTheme} from "../redux/actions";
import { describe, expect, it } from "@jest/globals";

describe( "InitialState reducer", () => {
    it( "should return the initial state", () => {
        expect(reducer( undefined, {} ) ).toEqual( {
            isLoggedIn: false,
            token: "",
            email: "",
            theme: "darkly"
        } );
    } );
    it( "should return is logined in true", () => {
        expect(
            reducer( [], {
                type: USER_AUTHENTICATED,
                payload: {isLoggedIn: true},
            } )
        ).toEqual( {
            isLoggedIn: true,
        } );
    } );
    it( "should return is logined out true", () => {
        expect(
            reducer( [], {
                type: USER_LOGGED_OUT,
                payload: {
                    isLoggedIn: false
                },
            } )
        ).toEqual( {
            isLoggedIn: false,
            token: "",
            loggedInUser: "",
            loggedInUserId: "",
        } );
    } );
    it( "should return is Theme", () => {
        const payload = {
            theme: "journal"
        }
        const expectedAction = {
            type: CHANGE_THEME,
            payload,
        };
        expect(ChangeTheme(payload)).toEqual(expectedAction);
    } );
} );
