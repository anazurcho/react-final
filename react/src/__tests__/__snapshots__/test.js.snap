// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queri test should return Sections 1`] = `undefined`;

exports[`queries test should return Sections 1`] = `undefined`;

exports[`query links test should return Sections 1`] = `undefined`;

exports[`query section test should return Sections 1`] = `undefined`;

exports[`query section test should return Sections 2`] = `undefined`;
